Signal/src/ViewControllers/OWSTableViewController.h

```diff
@@ -0,0 +1,56 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+#import <UIKit/UIKit.h>
+
+@class OWSTableItem;
+@class OWSTableSection;
+
+@interface OWSTableContents : NSObject
+
+@property (nonatomic) NSString *title;
+
+- (void)addSection:(OWSTableSection *)section;
+
+@end
+
+#pragma mark -
+
+@interface OWSTableSection : NSObject
+
+@property (nonatomic) NSString *title;
+
++ (OWSTableSection *)sectionWithTitle:(NSString *)title
+                                items:(NSArray *)items;
+
+- (void)addItem:(OWSTableItem *)item;
+
+@end
+
+#pragma mark -
+
+typedef NS_ENUM(NSInteger, OWSTableItemType) {
```

Comment:
Just curious, what other sort of thing do you envision being here besides an "action"?